### DISCUSSION TITLE
Partial types to amend shipped package types

### DISCUSSION THREAD

**Discussion by remcohaszing** [DefinitelyTyped/DefinitelyTyped#58268] [_Misc]
I’m using [`monaco-editor`](https://github.com/microsoft/monaco-editor). This package ships its own types, but not all of the public interface is typed. Although their readme states `monaco.d.ts` [_specifies the API of the editor_](https://github.com/microsoft/monaco-editor#installing), their [example code](https://github.com/microsoft/monaco-editor/blob/main/samples/browser-esm-webpack-small/index.js) uses untyped imports. There’s an open issue (https://github.com/microsoft/monaco-editor/issues/2450), but it’s not getting any attention. The code is generated from the Visual Studio Code repository, meaning that adding types is significantly more complex than just creating those `d.ts` files.

Is DefinitelyTyped the right place to add type definitions for this?

**Comment by martin-badin** [comment]
Hi. I suggest opening a PR into the source repository. DT doesn't preserve packages with its own types.
